src/Microsoft.Identity.Client/Platforms/net45/WebUIFactory.cs

```diff
@@ -35,15 +34,15 @@ internal class WebUIFactory : IWebUIFactory
     {
         private PlatformParameters parameters;
 
-        public IWebUI CreateAuthenticationDialog(IPlatformParameters inputParameters)
+        public IWebUI CreateAuthenticationDialog(IPlatformParameters inputParameters, UIBehavior behavior)
         {
             parameters = inputParameters as PlatformParameters;
             if (parameters == null)
             {
                 throw new ArgumentException("parameters should be of type PlatformParameters", nameof(parameters));
             }
 
-            if (parameters.UseHiddenBrowser)
+            if (behavior.Equals(UIBehavior.Never))
```

Comment:
E2E testing and manual apps are a much more expensive and unreliable way of testing, whereas testing the expected behaviour of this method using a unit test would be straightforward. It would be cheaper to unit test the platform-specific code where possible.